contributor-docs/adrs/adr-005-box-sx.md

```diff
@@ -0,0 +1,62 @@
+# ADR 005: Use Box as a building block for all other components
+
+## Status
+
+Proposed
+
+## Context
+
+In Primer React and consuming applications, we use many different patterns for creating React components. Two common patterns are:
+
+1. Creating components with styled-components
+
+    ```tsx
+    const Avatar = styled.img.attrs<StyledAvatarProps>(props => ({
+      height: props.size,
+      width: props.size
+    }))<StyledAvatarProps>`
+      display: inline-block;
+      overflow: hidden;
+      line-height: ${get('lineHeights.condensedUltra')};
+      border-radius: ${props => getBorderRadius(props)};
+      ${sx}
+    `
+    ```
+
+    [Show full code example →](https://github.com/primer/react/pull/2019/files?diff=split&w=0)
+
+2.  Creating components with Box
+
+    ```tsx
+    const Avatar: React.FC<AvatarProps> = ({size = 20, alt = '', square = false, sx = {}, ...props}) => {
+      const styles = {
+        display: 'inline-block',
+        overflow: 'hidden',
+        lineHeight: 'condensedUltra',
+        borderRadius: getBorderRadius({size, square})
+      }
+
+      return (
+        <Box
+          as="img"
+          alt={alt}
+          sx={merge(styles, sx as SxProp)} // styles needs to merge with props.sx
+          {...props}
+        />
+      )
+    }
+    ```
+
+    [Show full code example →](https://github.com/primer/react/pull/2019/files?diff=split&w=0)
+
+&nbsp;
+
+## Decision
+
+Prefer using method #2: Creating components with Box for the following reasons:
+
+- Better authoring experience with Typescript. With Box, we can improve the API and autocomplete for consuming primitives.
```

Comment:
@siddharthkp - can you link to your [research](https://github.com/github/primer/discussions/755#discussioncomment-2318144), which you mentioned in [this comment](https://github.com/primer/react/pull/2020/files#r849468656) from the ADR itself for easier discovery? I think it will help to elaborate why working with Primitives might be easier for anyone reading it later.